### PR TITLE
Add r-symphony

### PR DESCRIPTION
**Pull request by janjagusch**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/r-symphony/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    - r-uwot
+
+test:
+  commands:
+    - $R -e "library('symphony')"           # [not win]
+    - "\"%R%\" -e \"library('symphony')\""  # [win]
+
+about:
+  home: https://CRAN.R-project.org/package=symphony
+  license: GPL-3
```
Comment (janjagusch) [inline recipes/r-symphony/meta.yaml:85]:
```suggestion
  license: GPL-3.0-or-later
```

**Comment by mfansler** [comment]
Hi @janjagusch - I saw this hadn't been updated in a bit, so I went ahead and pushed some changes. Hope you don't mind!

**Comment by janjagusch** [comment]
> Hi @janjagusch - I saw this hadn't been updated in a bit, so I went ahead and pushed some changes. Hope you don't mind!

Hi @mfansler! Thanks for your help. Is there anything I could have done differently? I used https://github.com/bgruening/conda_r_skeleton_helper/commit/0c460dedff2e96548d526c06f1fe466769463160 to render the recipe.

Could you also confirm for `conda-forge/r` to be listed as a maintainer? Or should I remove that group from the maintainers list?

I also have a couple other PRs that probably require the same treatment:
- https://github.com/conda-forge/staged-recipes/pull/27309
- https://github.com/conda-forge/staged-recipes/pull/27308
- https://github.com/conda-forge/staged-recipes/pull/27306

**Comment by mfansler** [comment]
@janjagusch I think you did everything correct up through following the linter's advice (i.e., adding `stdlib` and adjust SPDX license). The linter does not respond to us (conda-forge/r), so please leave that in and one of us will review the recipe. Please undraft it if you're ready for a review!

Identifying the additional dependencies can be a tricky step and it is acceptable to troubleshoot in the PR (one doesn't have to solve it locally before submitting). This is aided by inspecting the logs for any errors, e.g.:

**linux-64** Linking Analysis:
```yaml
number of files: 31
Warning: rpath /home/conda/staged-recipes/build_artifacts/r-symphony_1724074049096/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/r-symphony_1724074049096/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
   INFO: sysroot: '/home/conda/staged-recipes/build_artifacts/r-symphony_1724074049096/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo', 'usr/share/locale/zh_TW/LC_MESSAGES/libc.mo', 'usr/share/locale/zh_CN/LC_MESSAGES/libc.mo', 'usr/share/locale/vi/LC_MESSAGES/libc.mo']'
  ERROR (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/libopenblas.so.0 found in ['conda-forge/linux-64::libopenblas==0.3.27=pthreads_hac2b453_1']
  ERROR (r-symphony,lib/R/library/symphony/libs/symphony.so): .. but ['conda-forge/linux-64::libopenblas==0.3.27=pthreads_hac2b453_1'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/R/lib/libR.so found in conda-forge/linux-64::r-base==4.3.3=h9967228_14
   INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==14.1.0=hc0a3c3a_0
   INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0
   INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-data.table==1.15.4=r43h5f06984_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-harmony==1.2.0=r43h08d816e_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rlang==1.1.4=r43ha18555a_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-magrittr==2.0.3=r43hb1dbf0f_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-matrix==1.6_5=r43he966344_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-purrr==1.0.2=r43hdb488b9_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-tibble==3.2.1=r43hdb488b9_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rcpp==1.0.13=r43h0d4f4ea_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): interpreted library (R) package conda-forge/noarch::r-rcolorbrewer==1.1_3=r43h785f33e_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-irlba==2.3.5.1=r43h0d28552_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rcpparmadillo==14.0.0_1=r43hb424bfc_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-uwot==0.1.16=r43h0d4f4ea_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rann==2.6.1=r43ha18555a_5 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): interpreted library (R) package conda-forge/noarch::r-ggplot2==3.5.1=r43hc72bb7e_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-tidyr==1.3.1=r43h0d4f4ea_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-dplyr==1.1.4=r43h0d4f4ea_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-class==7.3_22=r43hb1dbf0f_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

Namely, this shows that the resulting compiled library is linking against shared objects in the `host` environment without explicitly declaring that it needs them in the `run` environment. This can really vary package to package and does not have a programatic solution. It can be helpful to look at **the final linking step** done during the package compilation (breaks `\` added for readability):

```bash
x86_64-conda-linux-gnu-c++ -std=gnu++11 -shared -L$PREFIX/lib/R/lib \
  -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags \
  -Wl,--gc-sections -Wl,--allow-shlib-undefined \
  -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib \
  -o symphony.so RcppExports.o singlecellmethods.o utils.o \
  -llapack -lblas -lgfortran -lm -lquadmath -L$PREFIX/lib/R/lib -lR
```

So, we can see that it is *prepared*<sup>**1**</sup> to link to BLAS/LAPACK and some standard C++ and Fortran libs. The Fortran is often just a red herring I speculate is due to R package authors blindly copying templates for Makevars from examples (e.g., if I ask ChatGPT for a Makevars for an R package with C++ it will include `-lgfortran`). Anyway, that's why I added `libblas` and `liblapack`.

Looking at the **win-64** logs similarly indicated the need for the `libwinpthread`, but that one is slightly more complicated. One could search other PRs in the `staged-recipes` repository to find how it has been solved previously.

---

<sub>**[1]:** Having linker flag only requires that the library be present for the linker to run, but it doesn't actually mean the code itself has a reference to it. So it doesn't hurt to have extras of these at compile time even if the final library won't reference them. But for Conda Forge purposes, we only want to include in the `host` requirements the ones that are truly linked.</sub>

**Comment by janjagusch** [comment]
> @janjagusch I think you did everything correct up through following the linter's advice (i.e., adding `stdlib` and adjust SPDX license). The linter does not respond to us (conda-forge/r), so please leave that in and one of us will review the recipe. Please undraft it if you're ready for a review!
> 
> Identifying the additional dependencies can be a tricky step and it is acceptable to troubleshoot in the PR (one doesn't have to solve it locally before submitting). This is aided by inspecting the logs for any errors, e.g.:
> 
> **linux-64** Linking Analysis:
> 
> ```yaml
> number of files: 31
> Warning: rpath /home/conda/staged-recipes/build_artifacts/r-symphony_1724074049096/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/r-symphony_1724074049096/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
>    INFO: sysroot: '/home/conda/staged-recipes/build_artifacts/r-symphony_1724074049096/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo', 'usr/share/locale/zh_TW/LC_MESSAGES/libc.mo', 'usr/share/locale/zh_CN/LC_MESSAGES/libc.mo', 'usr/share/locale/vi/LC_MESSAGES/libc.mo']'
>   ERROR (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/libopenblas.so.0 found in ['conda-forge/linux-64::libopenblas==0.3.27=pthreads_hac2b453_1']
>   ERROR (r-symphony,lib/R/library/symphony/libs/symphony.so): .. but ['conda-forge/linux-64::libopenblas==0.3.27=pthreads_hac2b453_1'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
>    INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/R/lib/libR.so found in conda-forge/linux-64::r-base==4.3.3=h9967228_14
>    INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==14.1.0=hc0a3c3a_0
>    INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
>    INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0
>    INFO (r-symphony,lib/R/library/symphony/libs/symphony.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-data.table==1.15.4=r43h5f06984_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-harmony==1.2.0=r43h08d816e_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rlang==1.1.4=r43ha18555a_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-magrittr==2.0.3=r43hb1dbf0f_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-matrix==1.6_5=r43he966344_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-purrr==1.0.2=r43hdb488b9_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-tibble==3.2.1=r43hdb488b9_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rcpp==1.0.13=r43h0d4f4ea_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): interpreted library (R) package conda-forge/noarch::r-rcolorbrewer==1.1_3=r43h785f33e_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-irlba==2.3.5.1=r43h0d28552_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rcpparmadillo==14.0.0_1=r43hb424bfc_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-uwot==0.1.16=r43h0d4f4ea_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-rann==2.6.1=r43ha18555a_5 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): interpreted library (R) package conda-forge/noarch::r-ggplot2==3.5.1=r43hc72bb7e_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-tidyr==1.3.1=r43h0d4f4ea_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-dplyr==1.1.4=r43h0d4f4ea_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> WARNING (r-symphony): plugin library (R) package conda-forge/linux-64::r-class==7.3_22=r43hb1dbf0f_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> ```
> 
> Namely, this shows that the resulting compiled library is linking against shared objects in the `host` environment without explicitly declaring that it needs them in the `run` environment. This can really vary package to package and does not have a programatic solution. It can be helpful to look at **the final linking step** done during the package compilation (breaks `\` added for readability):
> 
> ```shell
> x86_64-conda-linux-gnu-c++ -std=gnu++11 -shared -L$PREFIX/lib/R/lib \
>   -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags \
>   -Wl,--gc-sections -Wl,--allow-shlib-undefined \
>   -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib \
>   -o symphony.so RcppExports.o singlecellmethods.o utils.o \
>   -llapack -lblas -lgfortran -lm -lquadmath -L$PREFIX/lib/R/lib -lR
> ```
> 
> So, we can see that it is _prepared_**1** to link to BLAS/LAPACK and some standard C++ and Fortran libs. The Fortran is often just a red herring I speculate is due to R package authors blindly copying templates for Makevars from examples (e.g., if I ask ChatGPT for a Makevars for an R package with C++ it will include `-lgfortran`). Anyway, that's why I added `libblas` and `liblapack`.
> 
> Looking at the **win-64** logs similarly indicated the need for the `libwinpthread`, but that one is slightly more complicated. One could search other PRs in the `staged-recipes` repository to find how it has been solved previously.
> 
> **[1]:** Having linker flag only requires that the library be present for the linker to run, but it doesn't actually mean the code itself has a reference to it. So it doesn't hurt to have extras of these at compile time even if the final library won't reference them. But for Conda Forge purposes, we only want to include in the `host` requirements the ones that are truly linked.

Thanks for the detailled response @mfansler! 👏 
I've marked the PR as ready for review.

**Review by mfansler — APPROVED** [review]
Looks good! Thanks for taking time to add this! ❤️

**Pull request merged**